### PR TITLE
Fixed inconsistent usage of keep_local and local_only kwarg in Help() function, its implementation and docs

### PR DESCRIPTION
**Pull request by hedger**
## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
Not sure why you believe this change is needed?
local_only, means only the local addoptions args are in the help.. what's the concern here?

**Comment by hedger** [comment]
@bdbaddog the concern is that documentation is inconsistent with actual behavior. 

Currently, [the docs](https://scons.org/doc/4.8.1/HTML/scons-man.html#f-Help) mention the following prototype for the Help() function:

> `env.Help(text, append=False, local_only=False)`

While actual implementation in https://github.com/SCons/scons/blob/4f3e3026bc455b3b47666567f273b2f84f932d32/SCons/Script/__init__.py#L254 uses the `keep_local` kwarg name.

I made the documentation consistent with the actual implementation - in order to not break any third-party code that relies on existing behavior. I also renamed the kwarg inside the implementation to avoid confusion.

**Comment by bdbaddog** [comment]
Seems like it would be simpler to just make everything `local_only` and not `keep_local`

also calling `Help()` in your SConscript calls SCons.SConscript.Help(), and not PrintHelp() (not directly).
Help() calls HelpFunction() which calls PrintHelp()

So the public interface is SCons.SConscript.Help(), and that's what's misdocumented.
(and mis doc-string'ed too I think)

Thoughts @mwichmann ?

File: CHANGES.txt [hunk 1]
```diff
@@ -49,6 +49,8 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
       cl.exe-compatible command line switches.
     - Some manpage cleanup for the gettext and pdf/ps builders.
     - Some clarifications in the User Guide "Environments" chapter.
+    - Fixed wrong naming in the documentation of "keep_local" parameter for
```
Comment (bdbaddog) [inline CHANGES.txt:52]: This should be changed to be attributed to you and not to mwichmann.
Please add a section with your name here ( it should be alphabetical by last name)

**Comment by bdbaddog** [comment]
Replacing this with #4670

**Pull request closed without merging**